datafusion/functions-nested/src/planner.rs

```diff
@@ -148,6 +148,16 @@ impl ExprPlanner for FieldAccessPlanner {
             // expr[idx] ==> array_element(expr, idx)
             GetFieldAccess::ListIndex { key: index } => {
                 match expr {
+                    // Special case for accessing map value with non-string values
+                    Expr::ScalarFunction(scalar_func) if is_map(&scalar_func) => {
+                        let Expr::Literal(name) = *index else {
+                            return plan_err!("index should be a literal");
+                        };
+                        Ok(PlannerResult::Planned(get_field(
```

Comment:
Could we use `map_extract` here?

Reply:
`map_extract` returns a list containing the value. Maybe can wrap it around `get_field` call and extract the value. But i am not sure, let me try it out.